Js課題３です

candle0403 opened the pull request.

candle0403 committed:
Merge pull request #1 from candle0403/master

JS課題１mergeデータです

Commit:
JS課題２です

いつもありがとうございます。
チェックよろしくお願いいたします

Commit:
JS課題２first-commit

いつもありがとうございます。
チェックよろしくお願いいたします

Commit:
fiestElement:delete修正

Commit:
末尾エラー文修正、リファクタリング

チェックお願いいたします！

Commit:
再リファクタリング、Wエラー文修正

Commit:
課題３first-commit

チエックよろしくお願いいたします！

Commit:
入力欄を都度空白へ

修正しました！

Commit:
タスク削除機能追加

チェックよろしくお願いいたします！

Commit:
押下した要素を特定し、削除

kouki-iwahara commented on JS課題３/js/main.js:
```diff
@@ -0,0 +1,51 @@
+				tr.appendChild(td4);
+				tbody.appendChild(tr);
+
+				td4.addEventListener("click", () => {
```
deleteTodoのような名前のtodoを削除する関数を作成し、クリックされたらその関数を呼び出すようにすると良いです。

現在、addTaskという関数に色々な目的の処理が混じっていますが、それぞれ目的別に関数を作成してみましょう。
今回の課題は大きく以下の関数に分けられます。
```
・todoを追加する
・todoを表示する
・状態ボタンを作成する
・削除ボタンを作成する
・todoを削除する
・todoの状態を切り替える
```
todoの状態を切り替える処理は次のステップで実装となりますので、このステップではtodo削除までの関数を実装してみましょう。

candle0403 replied:
ご指摘ありがとうございます。
・状態ボタンを作成する
・削除ボタンを作成する
・todoを削除する
こちらは、forEachメソッドを使って引数を利用して、tasksの中の配列を一気に書いているので、
関数に直すとスコープが発生してしまい、関数に分けることができず、実装できなくなってしまいます。
削除も、forEachメソッドを使って引数を利用しています。
どうしたら関数に分けることができるでしょうか・・・？

niganium replied:
関数の実行そのものはループ内ですることにはなるかと思いますが、関数の定義自体は他の箇所でもできるかと思うのでそれをすると見た目的にも役割がはっきりするのでは無いでしょうか？
一応本課題の要件は満たしているので再提出は不要ですがコードのチェックを依頼する場合は再度提出をお願いいたします。

candle0403 replied:
関数の実行はループ内でしています。
関数の定義も他の場所でしていますが、動きません。
試しに・todoを削除する
関数をclearTr(); 
で定義し、引数indexを使って動かしたいと思いましたが、関数の挙動が出ません。

どうしたら動くようになるでしょうか？
ご教授お願いいたします。

kouki-iwahara commented on JS課題３/js/main.js:
```diff
@@ -0,0 +1,51 @@
+				td.textContent = `${index}`;
+				tr.appendChild(td);
+				const td2 = document.createElement("td");
+				td2.textContent = `${task.comment}`;
```
task.commentには必ず文字列が入っているかと思いますが、
``${task.comment}``としているのは何か意図がありますか？

candle0403 replied:
配列の中のプロパティを使うにはテンプレートリテラルを使う癖がついていました・・・！！！
確かに、textContentは文字列ですし、task.commentには必ず文字列が入っていますね！
盲点でした。
ご指摘ありがとうございます！

kouki-iwahara commented on JS課題３/js/main.js:
```diff
@@ -0,0 +1,51 @@
+				const td = document.createElement("td");
+				td.textContent = `${index}`;
+				tr.appendChild(td);
+				const td2 = document.createElement("td");
```
変数に数値を使用すると抽象的な名前になってしまうので避けましょう。

candle0403 replied:
承知しました！わかりやすい変数の名前つけを意識していきます！
ありがとうございます。

kouki-iwahara commented on JS課題３/js/main.js:
```diff
@@ -0,0 +1,51 @@
+	const tbody = document.querySelector("tbody");
+
+	const tasks = [];
+	let i = -1;
```
どこにも使用されていない変数のようです。

kouki-iwahara commented on JS課題３/js/main.js:
```diff
@@ -0,0 +1,51 @@
+"use strict";
+{
+	const oneTask = document.getElementById("task");
```
JSは文字列を`""`ではなく`''`で囲む慣習がありますので修正しましょう。

candle0403 replied:
！これまですべて””を使ってしまっていました。。。
今後修正してまいります。
ご指摘感謝です
ありがとうございます！

Commit:
一部の関数化以外修正（質問しています！）

・シングルクオートに変更
・定数名修正（td２→commentTd　など）
・不要なテンプレートリテラル除去

niganium commented:
指摘点の修正を確認いたしました。問題ありません！
引き続き後続の作業をお願いいたします。

Commit:
Update main.js

Commit:
質問・削除ボタン関数が動かない

niganiumさんのいう通りに関数を宣言し、ループ内で動かしてみましたが、
削除ボタンが動きません。。。

どうしたらリファクタリングできるでしょうか？
よろしくお願いいたします。

Commit:
関数化完了

OK出ましたら３−３課題もすぐにプッシュいたします。
